src/DataStreams/PushingToViewsBlockOutputStream.h

```diff
@@ -1,20 +1,36 @@
 #pragma once
 
 #include <DataStreams/IBlockOutputStream.h>
+#include <Interpreters/QueryViewsLog.h>
 #include <Parsers/IAST_fwd.h>
 #include <Storages/IStorage.h>
 #include <Common/Stopwatch.h>
 
 namespace Poco
 {
 class Logger;
-};
+}
 
 namespace DB
 {
 
 class ReplicatedMergeTreeSink;
 
+struct ViewInfo
+{
+    const ASTPtr query;
+    StorageID table_id;
+    BlockOutputStreamPtr out;
```

Comment:
I think that BlockOutputStreamPtr is not actually an info

Reply:
You are right. It was already called ViewInfo though (I moved it outside the class to make it easier to share). Maybe ViewRuntimeData would make more sense?

Reply:
Yes, that's good.